### PR TITLE
My Home: Try Home Layout i1.

### PR DESCRIPTION
**Pull request by kwight**
This PR explores what changes we need to make for something like My Home Layout i1.

See: pbAPfg-g0-p2

#### Testing instructions
* Apply D41677-code and sandbox the API.
* Go to `/home/:site` with an prelaunch, launched and established site and observe the layout.

### CODE REVIEW

File: client/my-sites/customer-home/locations/primary/index.jsx [hunk 1]
```diff
@@ -2,29 +2,46 @@
...
+
+	const stats = cards.shift();
+
 	return (
 		<div className="primary">
+			<h2>{ translate( 'Stats at a glance' ) }</h2>
+			{ React.createElement( cardComponents[ stats ], {
+				checklistMode: stats === 'home-primary-checklist-site-setup' ? checklistMode : null,
+			} ) }
+			<h2>{ translate( 'Learn and grow' ) }</h2>
```
Comment (a8ci18n) [inline client/my-sites/customer-home/locations/primary/index.jsx:44]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Learn and grow with My Home' )` *ES Score: 6*

**Comment by kwight** [comment]
One of the first questions that comes up is, where do we get the section headings from (eg. "Learn and grow")? If we handle them in the client (as is done here), the translations are handled in Calypso, but we're kinda breaking our guiding idea that the client is just mindlessly rendering whatever the API passes to it (well, at least we're making assumptions in the client of what content we'll be rendering). The headings _could_ be translated on the server and sent by API based on a key to the section, but... that also feels weird and fragile. That would need more nested view arrays, I think:

Screen Shot 2020-04-09 at 3 23 53 PM

I can't decide if I like the idea of keeping the API responses just one level deep as they are now, and having the client would cut them up, like this PR does. It would depend on how much we anticipate the location sections changing; if we always have a stats card, it seems a premature optimization to make sure that can all be handled server-side.

**Comment by gwwar** [comment]
> I can't decide if I like the idea of keeping the API responses just one level deep as they are now, and having the client would cut them up, like this PR does.

My intuition is that keeping the client pretty simple would be ideal. We may need to break up the abstractions further. For example instead of just locations, do we specify the pages/tasks to render for a component. Let's play around a bit to see what feels best.

I'm also running into a similar problem for if we pick some sort of pager for educational cards. It could be that these turn into another "location/container"

image

**Comment by mmtr** [comment]
> It could be that these turn into another "location/container"

I like that. I think the API should treat each section with a header as a location.

```
{
  'alerts': [ ... ], // Congratulatory banners or site-wide notices.
  'up-next': [ ... ], // What to do up next (site setup checklist, current post-launch task).
  'stats': [ ... ], // Stats at a glance.
  'features': [ ... ], // Learn and grow.
  'management': [ ... ], // Manage your site.
}
```

Note that I deliberately not used current locations names so we can add the new layout to the endpoint without breaking existing behavior.

Calypso would decide then how to render each section: 
- Column mode. It's the default style, all cards displayed one below the other.
- Page mode. It'll combine the cards and display some sort of pager so the user can browse them. We'll use this mode on the educational cards.
- Stacked mode. It'll combine the cards and display them one behind the other. Dismissing / completing a card will show the one behind. We can use this mode on the notices.

File: client/my-sites/customer-home/locations/tasks/index.jsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ const cardComponents = {
 	'home-banner-legacy-stats-banners': StatsBanners,
 };
 
-const Upsells = ( { cards, primary, siteId, slug } ) => {
+const Tasks = ( { cards, primary, siteId, slug } ) => {
```
Comment (Aurorum) [inline client/my-sites/customer-home/locations/tasks/index.jsx:19]: Just a heads up that I started something for Tasks in #41018 :)

File: client/my-sites/customer-home/locations/features/index.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	'home-education-free-photo-library': FreePhotoLibrary,
+	'home-education-mastering-gutenberg': MasteringGutenberg,
+};
+
+const Features = ( { cards } ) => {
+	const translate = useTranslate();
+
+	return (
+		<>
+			<h2>{ translate( 'Learn and grow' ) }</h2>
```
Comment (a8ci18n) [inline client/my-sites/customer-home/locations/features/index.jsx:23]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Learn and grow with My Home' )` *ES Score: 6*

File: client/my-sites/customer-home/style.scss [hunk 4]
```diff
@@ -58,6 +58,10 @@
 		@include grid-template-columns( 12, 24px, 1fr );
 		grid-gap: 24px;
 	}
+
+	h2 {
+		font-family: 'Recoleta', serif;
```
Comment (gwwar) [inline client/my-sites/customer-home/style.scss:63]: We'll need to download the font, we can maybe check in to see how Gutenboarding is doing it. There are some performance considerations here as we do it, so we can maybe consult perfops on best practices.

**Comment by gwwar** [comment]
I think this is starting to come together nicely, I left some notes on the backend patch and asked a few clarifying questions for requirements in related thread.

Screen Shot 2020-04-10 at 4 03 01 PM

Some next steps/tasks in misc order (I'll file issues).
- [ ] Make sure Recoleta works, is somewhat okay perf wise for the download hit, and has a decent fallback for unsupported languages
- [ ] Add two extra support videos to Mastering the Block Editor ( Block Editor Overview / Layouts/Block-Patterns)
- [ ] Update the educational container to have a temporary dot pager so we can see both mastering the block editor and free photo library
- [ ] File PR to Gutenberg to break out the pager from the guide component so we can reuse this in Calypso https://wordpress.github.io/gutenberg/?path=/story/components-guide--default
- [ ] Create a new stats component for the new graph view. We can then deprecate and delete the old stats card component after we've switched over views.

File: client/my-sites/customer-home/locations/secondary/learn-grow/index.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	'home-education-free-photo-library': FreePhotoLibrary,
+	'home-education-mastering-gutenberg': MasteringGutenberg,
+};
+
+const LearnGrow = ( { cards } ) => {
+	const translate = useTranslate();
+
+	return (
+		<>
+			<h2>{ translate( 'Learn and grow' ) }</h2>
```
Comment (a8ci18n) [inline client/my-sites/customer-home/locations/secondary/learn-grow/index.jsx:26]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Learn and grow with My Home' )` *ES Score: 6*

File: client/my-sites/customer-home/cards/education/free-photo-library/index.jsx [hunk 6]
```diff
@@ -3,76 +3,39 @@
...
-					<img
-						className="free-photo-library__demonstration-image"
-						src={ freePhotoLibraryVideoPrompt }
-						alt={ translate( 'Free Photo Library demonstration' ) }
-					/>
-				</button>
+		<EducationalCard
+			header={ translate( 'The WordPress.com free photo library' ) }
+			text={ translate(
+				'Our free photo library integrates your site with over 40,000 beautiful copyright-free photos to create stunning designs.'
```
Comment (a8ci18n) [inline client/my-sites/customer-home/cards/education/free-photo-library/index.jsx:26]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'The WordPress.com Free Photo Library integrates your site with beautiful copyright-free photos to create stunning designs.' )` *ES Score: 8*

File: client/my-sites/customer-home/locations/secondary/learn-grow/index.jsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+	'home-education-gutenberg': MasteringGutenberg,
+};
+
+const LearnGrow = ( { cards } ) => {
+	const translate = useTranslate();
+
+	return (
+		<>
+			<h2 className="learn-grow__heading customer-home__section-heading">
+				{ translate( 'Learn and grow' ) }
```
Comment (a8ci18n) [inline client/my-sites/customer-home/locations/secondary/learn-grow/index.jsx:28]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Learn and grow with My Home' )` *ES Score: 6*

**Comment by gwwar** [comment]
@kwight I think we can close this one out? Please reopen if we still need this

**Pull request closed without merging**